.github/workflows/pytest.yml

```diff
@@ -105,31 +109,31 @@ jobs:
           path: coverage.xml
 
   ci-integration:
-    runs-on:
-      group: large-runner-group
-      labels: ubuntu-22.04-4core
+    needs:
+      - ci-unit
```

Comment:
What's your thought on including this dependency between jobs? Is it just to avoid running the long, expensive `ci-integration` tests if we already know something is wrong?